docs/devel/networking.md

```diff
@@ -326,3 +326,51 @@ Chain POSTROUTING (policy ACCEPT)
 target     prot opt source               destination
 MASQUERADE  all  --  10.0.2.2             anywhere
 ```
+
+### Masquerade binding using IPv6 addresses
+The masquerade binding mechanism is currently the only binding mechanism which
+accepts IPv6 addresses.
+
+It operates in exactly the same way as in IPv4, and follows the same goals:
+configure one to one NAT. As in it's IPv4 counter-part, the pods are reached
+via their IPv6 pod addresses.
+
+NAT is configured in the exact same way, but using ip6tables, or the `ipv6-nat`
+nftable table. Please refer to the tables below to visualize how NAT for IPv6
+addresses is accomplished in KubeVirt.
+
+```
+Chain PREROUTING (policy ACCEPT)
+target               prot opt source               destination
+KUBEVIRT_PREINBOUND  all      anywhere             anywhere
+
+Chain INPUT (policy ACCEPT)
+target     prot opt source               destination
+
+Chain OUTPUT (policy ACCEPT)
+target     prot opt source               destination
+DNAT       tcp      anywhere             localhost            tcp dpt:http to:fd10:0:2::2
+
+Chain POSTROUTING (policy ACCEPT)
+target     prot opt source               destination
+MASQUERADE  all      fd10:0:2::2          anywhere
+KUBEVIRT_POSTINBOUND  all      anywhere   anywhere
+
+Chain KUBEVIRT_POSTINBOUND (1 references)
+target     prot opt source               destination
+SNAT       tcp      anywhere             anywhere             tcp dpt:http to:fd10:0:2::1
+
+Chain KUBEVIRT_PREINBOUND (1 references)
+target     prot opt source               destination
+DNAT       tcp      anywhere             anywhere             tcp dpt:http to:fd10:0:2::2
+```
+
+It is important to refer that when IPv6 masquerade binding is set in place,
+KubeVirt also configures NAT for IPv4 addresses - i.e. the IPv4 traffic is also
+forwarded into the VM instance. Despite that, the only IP address reported is
```

Comment:
@phoracek I wonder about your PR #3483 ; don't we want the VMI / pod to report both IPv4 & IPv6 addresses ? 

I get this when running it on the kind ipv6 provider:

```yaml
  phase: Running
  podIP: fd00:10:244::22
  podIPs:
  - ip: fd00:10:244::22
```

Reply:
On the API, we want to report externally (from the network) accessible IPs. Since you used masquerade binding, both internal IPs are not accessible and thus not reported. With masquerade, we report only PodIP (once there is dual-stack support, we would report all PodIPs.

If you use bridge binding and have guest agent available, it would report all the IPs found in guest.